pkg/agent/controller/egress/egress_controller_test.go

```diff
@@ -869,10 +869,11 @@ func TestSyncEgress(t *testing.T) {
 			if tt.newLocalIPs != nil {
 				c.localIPDetector = &fakeLocalIPDetector{localIPs: tt.newLocalIPs}
 			}
-			assert.NoError(t, wait.Poll(time.Millisecond*100, time.Second, func() (done bool, err error) {
-				egress, _ := c.egressLister.Get(tt.newEgress.Name)
-				return reflect.DeepEqual(egress, tt.newEgress), nil
-			}))
+			assert.NoError(t, wait.PollUntilContextTimeout(context.Background(), time.Millisecond*100, time.Second, true,
```

Comment:
Should the argument `immediate` be `true` to replace wait.Poll?

Reply:
After checking the doc and some examples in Kubernetes repo, I believe the argument `immediate` should be set to `false` when using `wait.Poll`. I will fix these functions, thanks for the reminder.